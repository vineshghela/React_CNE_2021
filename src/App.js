// import MyFirstComponent from './Components/01-intro/MyFirstComponent'
import MySecondComponent from './Components/01-intro/MySecondComponent'
import MyFooter from './Components/02-Parent-Child/MyFooter'
import Standard from './Components/02-Parent-Child/Standard'
import Parent from './Components/03-Props/01-Simple/Parent'
import PropWithDifferentTypes from './Components/03-Props/02-intermediate/PropWithDifferentTypes'
import EndContactCard from './Components/03-Props/03-Contact-Card/EndContactCard '
import FirstContactCard from './Components/03-Props/03-Contact-Card/WIPFirstContactCard'
import Blog from './Components/03-Props/Static-data/CommentData/Blog'
import User from './Components/03-Props/Static-data/PeopleData/User'
import Strap from './Components/03-Props/Strap'
import ClassClock from './Components/04-Lifecycle/ClassClock'
import ClockFunctional from './Components/04-Lifecycle/ClockFunctional'
// import Counter from './Components/05-States/AddingState/Counter'
import DifferentStates from './Components/05-States/DifferentStates'
import LetsSearch from './Components/05-States/Tasks/01/LetsSearch'
import HomePage from './Components/05-States/Tasks/02/HomePage'
import ControlledInput from './Components/06-Forms/ControlledInput'
import UncontrolledInput from './Components/06-Forms/UncontrolledInput'
// import SearchableList from './Components/07-LiftingStates/01-Search/SearchableList'
import SearchableList from './Components/07-LiftingStates/02-Checked/SearchableList'
import Clock from './Components/08-Hooks/Clock'
import Counter from './Components/08-Hooks/Counter'
import Director from './Components/09-Routing/Director'
import Tesco from './Components/10-Data-Requests/Tesco'

const App=()=> {
  return (
    <div className="App">
      <h1>React - CNE 2021</h1>
      {/* <MyFirstComponent/> */}
      {/* <MySecondComponent/> */}
      {/* <Standard/> */}
      {/* <MyFooter/> */}
      {/* <Parent/> */}
      {/* <PropWithDifferentTypes/> */}
      {/* <FirstContactCard/> */}
      {/* <EndContactCard/> */}
      {/* <Blog/> */}
      {/* <User/> */}
      {/* <Strap/> */}
      {/* <ClassClock/> */}
      {/* <ClockFunctional/> */}
      {/* <DifferentStates/> */}
      {/* <Counter/> */}
      {/* <UncontrolledInput/> */}
      {/* <ControlledInput/> */}
      {/* <LetsSearch/> */}
      {/* <HomePage/> */}
      {/* <SearchableList/> */}
      {/* <Clock/>
      <Counter/> */}
      {/* <Director/> */}

      <Tesco/>


    </div>
  );
}

//ONLY ONE DEFAULT EXPORT PER FILE
export default App;
