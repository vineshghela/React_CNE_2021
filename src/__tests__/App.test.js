import { render, screen } from '@testing-library/react';
import App from '../App';

import '@testing-library/jest-dom';
import {expect, it} from '@jest/globals';
import {create} from 'react-test-renderer';

it('should find a heading with React', () => {
  render(<App />);
  const linkElement = screen.getByText(/React/i);
  expect(linkElement).toBeInTheDocument();
});

it(`It should take a snapshot every time it's redered`,()=>{
  const snapShot = create(<App/>).toJSON()
  console.log(snapShot);
  expect(snapShot).toMatchSnapshot();
  //if it does not have the initial snapshot then during line 17 it will snapshot
  //only ever do it on the first time its ran.
  //Every test that run from this point on wards will use the snapshot it has as a reference 
})
