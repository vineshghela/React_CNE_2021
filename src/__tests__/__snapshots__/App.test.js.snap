// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should take a snapshot every time it's redered 1`] = `
<div
  className="App"
>
  <h1>
    React - CNE 2021
  </h1>
  <p>
    Please wait.... we are getting your request
  </p>
  <div
    className="spinner-border text-primary"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-border text-Danger"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-border text-info"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-border text-primary"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
